cheatsheets/Java_Security_Cheat_Sheet.md

```diff
@@ -507,10 +507,10 @@ other log viewing/analysis tools that don't expect the log data to be pre-HTML e
 
 - **Never, ever write your own cryptographic functions.**
 - Wherever possible, try and avoid writing any cryptographic code at all. Instead try and either use pre-existing secret management solutions or the secret management solution provided by your cloud provider. For more information, see the [OWASP Secrets Management Cheat Sheet](/Secrets_Management_Cheat_Sheet).
+- If you cannot use a pre-existing secret management solution, try and use a trusted and well known implementation library rather than using the libraries built into JCA/JCE as it is far too easy to make cryptographic errors with them.
 - Make sure your application or protocol can easily support a future change of cryptographic algorithms.
 - Use your package manager wherever possible to keep all of your packages up to date. Watch the updates on your development setup, and plan updates to your applications accordingly.
-- Wherever possible, use a trusted and well known implementation library rather than using the libraries built into JCA/JCE as it is far too easy to make cryptographic errors with them.
-- We will show examples below based on Google Tink which is an opinionated library for performing encryption operations.
+- We will show examples below based on Google Tink which is an [opinionated](https://www.baeldung.com/cs/opinionated-software-design#opinionated-design) library for performing encryption operations.

```

Comment:
I still _really_ dislike the word "opinionated" here, regardless of the link defining it. In part, that's because of the normal use of the word, and in part, that link starts off with "Generally, we can see the opinionated design as an **_oracle_** that indicates the right way to accomplish tasks" (emphasis mine). The word "oracle" has generally negative connotations in the cryptography world (e.g., as in "padding oracle attack").

**Suggestion**: Change the sentence
> We will show examples below based on Google Tink which is an opinionated library for performing encryption operations.

to something like

> We will show examples below based on Google Tink, which is a  library created by cryptography experts for using cryptography safely (in the sense of minimizing foolish coding mistakes for regular developers).

Yes, it's more words and perhaps not as pithy, but I think it better captures the design intent of its creators. The whole point is while this is "opinionated" in the sense of the Baeldung reference you provided, to me it better explains why a developer should be more apt to trust it than say JCA/JCE which doesn't restrict implementation choices but is very much an underlying cause of the mess we are in with cryptography.

Reply:
Addressed in  #1160